File: leetcode_75/Array_String/python/443.py
from typing import List


class Solution:
    def compress(self, chars: List[str]) -> int:
        counter = 1
        counter_index = 1
        start_index = 0
        for i in range(1, len(chars)):
            if chars[i] != chars[start_index]:
                if counter > 1:
                    for c in str(counter):
                        chars[counter_index] = c
                        counter_index += 1
                start_index = i
                counter = 1
                counter_index = i + 1
            else:
                chars[i] = ""
                counter += 1

        if not chars[-1]:
            for c in str(counter):
                chars[counter_index] = c
                counter_index += 1

        ind = 0
        while ind < len(chars):
            if not chars[ind]:
                del chars[ind]
                ind -= 1
            ind += 1
            
        print(chars)
        return len(chars)


solution = Solution()
print(solution.compress(["a", "b", "b", "b", "b",
      "b", "b", "b", "b", "b", "b", "b", "b"]))

File: leetcode_75/Array_String/cpp/443.cpp
#include <iostream>
#include <vector>
#include <string>
#include <algorithm>

using namespace std;

class Solution
{
public:
    int compress(vector<char> &chars)
    {
        int counter = 1;
        int counter_index = 1;
        int start_index = 0;
        for (int i = 1; i < chars.size(); ++i)
        {
            if (chars[i] != chars[start_index])
            {
                if (counter > 1)
                {
                    for (auto c : to_string(counter))
                    {
                        chars[counter_index] = c;
                        counter_index++;
                    }
                }
                start_index = i;
                counter = 1;
                counter_index = i + 1;
            }
            else
            {
                chars[i] = ' ';
                counter++;
            }
        }

        if (chars[chars.size() - 1] == ' ')
        {
            for (auto c : to_string(counter))
            {
                chars[counter_index] = c;
                counter_index++;
            }
        }

        for (auto c : chars)
        {
            cout << c << " ";
        }
        cout << endl;

        int ind = 0;
        while (ind < chars.size())
        {
            if (chars[ind] == ' ')
            {
                chars.erase(chars.begin() + ind);
                ind--;
            }
            ind++;
        }

        for (auto c : chars)
        {
            cout << c << " ";
        }
        cout << endl;

        return chars.size();
    }
};